sycl/doc/extensions/SPIRV/SPV_INTEL_optimization_hints.asciidoc

```diff
@@ -0,0 +1,183 @@
+
+Name Strings
+------------
+
+SPV_INTEL_optimization_hints
+
+Contact
+-------
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+Contributors
+------------
+
+- Dmitry Sidorov, Intel
+- Alexey Sachkov, Intel
+- Alexey Sotkin, Intel
+- Ben Ashbaugh, Intel
+
+Notice
+------
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+Status
+------
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a
+feature for review and community feedback. When the feature matures, this
+specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are
+subject to change they are not intended to be used by shipping software
+products.
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-04-27
+| Revision           | E
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds *AssumeTrueINTEL* and *OpExpectINTEL* instructions that are
+appropriately mapped on ‘llvm.assume’ and ‘llvm.expect’ intrinsics. That is
+required to save optimization hints through LLVM IR to SPIR-V transformation.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_optimization_hints"
+----
+
+New Capabilities
+----------------
+This extension introduces a new capability:
+
+----
+OptimizationHintsINTEL
+----
+
+New Instructions
+----------------
+Instructions added under the *OptimizationHintsINTEL* capability:
+
+----
+ExpectINTEL
+AssumeTrueINTEL
+----
+
+Token Number Assignments
+------------------------
+[width="45%",cols="30,15"]
+|===============================
+| OptimizationHintsINTEL | 5629
+| AssumeTrueINTEL        | 5630
+| OpExpectINTEL          | 5631
+|===============================
+
+Modifications to the SPIR-V Specification, Version 1.5
+------------------------------------------------------
+
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding these rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Depends On
+| 5629 | *OptimizationHintsINTEL* +
+Allow to use AssumeTrueINTEL and ExpectINTEL instructions |
+|====
+--
+
+Instructions
+~~~~~~~~~~~~
+
+In section 3.32.1, Miscellaneous Instructions, add a new instruction:
+
+[cols="3", width="100%"]
+|=====
+2+^|*OpAssumeTrueINTEL* +
+The instruction allows the optimizer to assume that the provided _Condition_ is
+always true whenever the control flow reaches the instruction call. If the
+_Condition_ is violated during execution, the behavior is undefined.
+
+The type of _Condition_ shall be _Boolean_.
```

Comment:
```suggestion
The type of _Condition_ must be _Boolean_.
```

Reply:
I disagree. Usually in specification/requirement docs `shall` has a power of `must`, so it's a strong requirement.

Reply:
Current version of the [SPIRV spec](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html) contains:
 * "shall" 0 times
 * "must" 1390 times

Reply:
Applied. If you don't mind I would leave the revision version unchanged :)